README.md

```diff
@@ -39,7 +93,34 @@ It is also possible for Marsha to infer the data type from CSV file
 
 ### Functions
 
-Functions are the bread and butter of Marsha and can easily define transformations between different data types. They are defined by a name, input, output (if not void), a description of the requirements, and a list of samples. The examples are imperative to be able to generate the test suite for the generated software which helps ensure accuracy. Function samples should look a lot like the requirements given to a software engineer in English, but using shorthand mathematical notation.
+Functions are the bread and butter of Marsha and can easily define transformations between different data types. There are three sections to a Marsha function: the declaration, the description, and the examples.
+
+The declaration is a Markdown section prefixed with `func`, then followed by a name, parenthesis containing the input type(s), and finally a colon followed by the output type. The name must be a single word, but the types don't need to be classic software types, or even the explicit data types defined above. They can themselves be simple descriptions of what the type is meant to be. Eg,
+
+```md
+# func get_employee_skills(list of EmployeesByDepartment, list of DepartmentSkills): list of EmployeeSkills
+```
+
+The next section is the description of the function. Here you explain what the function should do. Being more explicit here will reduce variability in the generated output and improve reliability in behavior, but it's up to you just how explicit you will be and how much you leave to the LLM to figure out. This is similar to declarative languages like SQL and HTML where there are defaults for things you do not specify. Eg,
```

Comment:
I think I do not fully follow the last sentence

Reply:
I guess I could include an example, hmm.

```suggestion
The next section is the description of the function. Here you explain what the function should do. Being more explicit here will reduce variability in the generated output and improve reliability in behavior, but it's up to you just how explicit you will be and how much you leave to the LLM to figure out. This is similar to declarative languages like SQL and HTML where there are defaults for things you do not specify, like the sort order of `select` statements or the default styling of a `<div>`. Eg,
```